containers/init/clone/clone.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# Copyright 2020 gRPC authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+git clone --recursive $CLONE_REPO /src/workspace
```

Comment:
one more nit: for some repositories, always doing a recursive clone might be actually a waste, not all the builds will need the submodules (e.g. when you build with bazel, you don't need them). So in the future we might wanna make the `--recursive` optional.  Totally fine to leave as is for now, just wanted to point this out.